proposed/2022/Package-Source-Mapping-VS-Options.md

```diff
@@ -0,0 +1,145 @@
+# Package Source Mapping in Visual Studio Options Dialog
+
+* Start Date: 06-10-2022
+* Authors: Ella McNally
+* Issue: https://github.com/NuGet/Home/issues/11363
+
+## Summary
+
+Currently there is no support for package source mapping in the Visual Studio options dialog. I will be creating a feature where users can pin default sources and choose a specific source(s) for certain package source mappings from sources already provided in package sources page. This feature is a portion of what was discussed in a previous spec. https://github.com/NuGet/Home/blob/dev/implemented/2021/PackageSourceMapping.md
+
+## Motivation 
+
+Adding support for package source mapping in the VS options dialog will allow user to pin sources and add/remove package source mappings with more ease.
+
+## Explanation
+
+### Functional Explanation
+
+#### Package Source Mappings Page 
+
+When the user opens the package source mapping page in the options dialog, they will see a list of all the package source mappings they have already configured and the filepath to the solution level config file. There is a hierarchy of NuGet configs levels: machine, user, and solution. Package source mappings will be read from all of the levels, but only written to the solution level config. This page will have three buttons: _Add_ , _Remove_, and _Clear_.
+
+* Add: Brings user to a pop-up where they can add package source mappings to their project.
+
+* Remove: Lets user remove selected package source mapping from project.
+
+* Clear: Lets user clear all package source mappings from current project.
+
+The popup when the user clicks the add button will have a textbox where a user can type in a package ID. All sources that are defined in package sources page of VS options dialog will appear in a `ComboBox` below. The user checks a `CheckBox` for the sources they would like to use for that package ID. 
+
+The add window has two buttons: _Add_ and _Close_.
+
+* Add: Add button adds package source mapping to project only using selected sources, and the pop-up will disappear. 
+
+* Close: Will close pop-up without adding any package source mappings.
+
+Below are the mockups for the package source mappings page. The top left is the package source mappings page, and the bottom left is the popup that appears when the user clicks the add button.

```

Comment:
How are the mappings going to be sorted? 

I think it allowing them to be sortable by both source name and pattern might be a neat feature. 
We did find that some people do prefer the mental model of source first, but others did like the pattern first approach instead. 
This might require unmerging some options.

In my opinion, the default should be sorted by source first, mostly because that's what the configuration model looks like. 

If we wanted to, we could even consider grouping them under a source node instead the same way the configuration does it.